### DISCUSSION TITLE
GIt commit message format

### DISCUSSION THREAD

**Discussion by metalalive** [Game-as-a-Service/Pandemic-Reign-of-Cthulhu#7] [Ideas]
Hi @wingtkw , @teds-lin 
I dunno whether you would use automation tool to format the commit message,
It could be more consistent to write commit message in following structure :
```
[#github-issue-number]  {commit-type} : {short-description}
``` 
Where :
- `github-issue-number`  maps to ID of specific GitHub issue
- `commit-type` can be `feature`, `bugfix`, `refactor`, `doc` , which indicates the category of the commit
- `short-description` is the intent of the commit

For example :
```
[#49]  feature : assign summoning card to active player at use-case layer
[#15]  bugfix : correct sanity dice result
[#10]  build : update build script for compiling python C extension module
[#11]  doc : modify API endpoint for setting game difficulty
```

If you like the idea above ,
please press thumb-up emoji then let's start following the commit structure from now on,
 thanks

### Reference
- [Commit Message Guidelines](https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53)
- [Writing Meaningful Git Commit Messages](https://medium.com/@menuka/writing-meaningful-git-commit-messages-a62756b65c81)

**Comment by KalpasWang** [comment]
thanks for your advice! learned a lot from you.🙏
